go1.25 release notes (#1800)
microsoft/go: docs/go1.25.md

# Microsoft build of Go 1.25 release notes

After the release of 1.25, 1.23 is no longer supported, per the [Go release policy](https://go.dev/doc/devel/release).

## System-provided cryptography enabled by default

The Microsoft build of Go 1.25 enables the `systemcrypto` experiment by default:

- **Linux:** Uses OpenSSL (requires `cgo`)

> - **Windows:** Uses CNG (does *not* require `cgo`)

Reviewer: cgo is an actual thing.

```suggestion
- **Linux:** Uses OpenSSL (requires cgo)
- **Windows:** Uses CNG (does *not* require cgo)
```

> - **macOS:** Systemcrypto backend remains in preview

Reviewer: In the review for https://devblogs.microsoft.com/go/microsoft-go-defaults-to-system-crypto/ I got rid of "`systemcrypto` backend" because we're sticking with "backend" being one of OpenSSL/CNG/CryptoKit, and `systemcrypto` just being an experiment that selects one.

Also, importantly, it isn't on by default unlike the other two. Splitting to a note seems like it helps keep it separate.

```suggestion

> [!NOTE]
> **macOS:** system-provided crypto backend remains in preview and is not enabled by default.
```

> This aligns with Microsoft's internal security and compliance policies. You may need to take action if your builds rely on Linux without `cgo`, use distroless containers, or have cross-distro deployment requirements.

Reviewer: ```suggestion
This aligns with Microsoft's internal security and compliance policies. You may need to take action if your builds rely on Linux without cgo, use distroless containers, or have cross-distro deployment requirements.
```

> To opt out of systemcrypto, set `GOEXPERIMENT=nosystemcrypto`.

Reviewer: ```suggestion
To opt out of systemcrypto, set the `GOEXPERIMENT` environment variable to include `nosystemcrypto`.
```

Same issue for non-`sh`-ers as other env, and say "include" to account for comma separated lists.

For full documentation, see the [Microsoft build of Go FIPS guide](https://github.com/microsoft/go/blob/microsoft/release-branch.go1.25/eng/doc/fips/UserGuide.md).

## Telemetry collection enabled

> Go 1.25 introduces opt-out telemetry collection to help us prioritize features, identify performance bottlenecks, and understand real-world developer workflows. All telemetry is anonymized and handled in accordance with [Microsoft's privacy policies](https://privacy.microsoft.com/privacystatement).

Reviewer: ```suggestion
The Microsoft build of Go 1.25 introduces opt-out telemetry collection to help us prioritize features, identify performance bottlenecks, and understand real-world developer workflows. All telemetry is anonymized and handled in accordance with [Microsoft's privacy policies](https://privacy.microsoft.com/privacystatement).
```

# Microsoft build of Go 1.25 release notes

After the release of 1.25, 1.23 is no longer supported, per the [Go release policy](https://go.dev/doc/devel/release).

## System-provided cryptography enabled by default

The Microsoft build of Go 1.25 enables the `systemcrypto` experiment by default:

- **Linux:** Uses OpenSSL (requires cgo)
- **Windows:** Uses CNG (does *not* require cgo)

> [!NOTE]
> **macOS:** system-provided crypto backend remains in preview and is not enabled by default.

This aligns with Microsoft's internal security and compliance policies. You may need to take action if your builds rely on Linux without cgo, use distroless containers, or have cross-distro deployment requirements.

To opt out of systemcrypto, set the `GOEXPERIMENT` environment variable to include `nosystemcrypto`.

> For full documentation, see the [Microsoft build of Go FIPS guide](https://github.com/microsoft/go/blob/microsoft/release-branch.go1.25/eng/doc/fips/UserGuide.md).

Reviewer: Missed that this is the wrong doc, and it should also always point at the main branch for the maintained copy.

```suggestion
For full documentation, see the [Microsoft build of Go FIPS guide](https://github.com/microsoft/go/blob/microsoft/main/eng/doc/fips/README.md).
```

# Microsoft build of Go 1.25 release notes

After the release of 1.25, 1.23 is no longer supported, per the [Go release policy](https://go.dev/doc/devel/release).

> ## 🚀 Summary

Reviewer 2: Worth a WARN or something along those lines?

> The Microsoft build of Go 1.25 introduces two major changes:

Reviewer: I don't think a summary is necessary, but if we have one, the user already expects to see it here, no need for the preamble.

- **System-provided cryptography is now enabled by default** via the `systemcrypto` experiment.

> - **Telemetry collection is now enabled by default** to help us improve the Microsoft Go toolchain based on real-world usage.

Reviewer: If we have a summary (I don't think we need one for this release) I think it should be styled as a clickable table of contents, and it should focus on making the change crystal clear without necessarily including a defense/rationale.

- **Distroless/minimal containers:** Base images without OpenSSL or glibc will not support systemcrypto. Either switch base images or opt out.
- **Cross-distro deployment on Linux:** `cgo` introduces a glibc dependency. Ensure compatibility between build and deployment systems or build on the oldest supported OS version.

To opt out of systemcrypto, set `GOEXPERIMENT=nosystemcrypto`.

For full documentation, see the [Microsoft build of Go FIPS guide](https://github.com/microsoft/go/blob/microsoft/release-branch.go1.25/eng/doc/fips/UserGuide.md).

## 📊 Telemetry Collection Enabled

Go 1.25 introduces opt-out telemetry collection in the Microsoft build of Go to help us:

- Prioritise features based on usage patterns  
- Identify performance bottlenecks  
- Understand real-world developer workflows  
- Make informed decisions about future improvements  

### What data is collected?

We collect anonymised usage and performance data from the Microsoft Go toolchain. This data is handled in accordance with [Microsoft’s privacy policies](https://privacy.microsoft.com/privacystatement) and applicable laws.

> ### 🔒 Privacy and control

Reviewer: Up until now emoji are associated with a `##`, not a `###`, so this is a bit confusing.

On the other hand, I think subsections probably indicate the doc is too complex.

- All telemetry is **anonymised**
- You have full control and can **opt out at any time**
- Telemetry is **independent of official Go telemetry**

### 🚫 How to opt out

> To disable telemetry, set `MS_GOTOOLCHAIN_TELEMETRY_ENABLED=0`.

Reviewer: Not clear what this means to someone who isn't used to working in a `sh`.
